test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -232,51 +253,52 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, cluster *
 		return ctrl.Result{}, nil
 	}
 
-	// Make sure bootstrap data is available and populated.
-	if machine.Spec.Bootstrap.DataSecretName == nil {
-		if !util.IsControlPlaneMachine(machine) && !conditions.IsTrue(cluster, clusterv1.ControlPlaneInitializedCondition) {
-			log.Info("Waiting for the control plane to be initialized")
-			conditions.MarkFalse(dockerMachine, infrav1.ContainerProvisionedCondition, clusterv1.WaitingForControlPlaneAvailableReason, clusterv1.ConditionSeverityInfo, "")
+	if !machinePoolOwned {
```

Comment:
I think that we are missing a big opportunity here.

Previously the DockerMachinePool controller had to reimplement all the logic to create and bootstrap containers because it was not possible to leverage on this logic.

Now that also DockerMachinePool creates DockerMachine we can (and we should) drop all the duplicate code in https://github.com/kubernetes-sigs/cluster-api/blob/8b15bc070b263b7b73d50a0a7b7ed8d20b120fc8/test/infrastructure/docker/exp/internal/docker/nodepool.go#L248 (and probably in other places) and let this controller be the only controller working on docker machines.

In other words, the DockerMachinePool controller should just orchestrate DockerMachines creation and deletion, a let the DockerMachine controller do the heavy lifting (Afaik the only delta will be in how to fetch the bootstrap token).

Reply:
Gotcha. So in this case, do we want to remove the logic like `external.Create()` from nodepool.go as well? If that's the case, I'm wondering what we want to keep in nodepool.go.